nodes/node_balancer/cmd/db.go

```diff
@@ -0,0 +1,108 @@
+package cmd
+
+import (
+	"database/sql"
+	"fmt"
+
+	configs "github.com/bugout-dev/moonstream/nodes/node_balancer/configs"
+
+	_ "github.com/lib/pq"
+)
+
+var (
+	databaseClient DatabaseClient
+)
+
+type DatabaseClient struct {
+	Client *sql.DB
+}
+
+// Establish connection with database
+func InitDatabaseClient() error {
+	db, err := sql.Open("postgres", configs.MOONSTREAM_DB_URI_READ_ONLY)
+	if err != nil {
+		return fmt.Errorf("DSN parse error or another database initialization error: %v", err)
+	}
+
+	// Set the maximum number of concurrently idle connections,
+	// by default sql.DB allows a maximum of 2 idle connections.
+	db.SetMaxIdleConns(configs.MOONSTREAM_DB_MAX_IDLE_CONNS)
+
+	// Set the maximum lifetime of a connection.
+	// Longer lifetime increase memory usage.
+	db.SetConnMaxLifetime(configs.MOONSTREAM_DB_CONN_MAX_LIFETIME)
+
+	databaseClient = DatabaseClient{
+		Client: db,
+	}
+
+	return nil
+}
+
+type Block struct {
+	BlockNumber      uint64      `json:"block_number"`
+	Difficulty       uint64      `json:"difficulty"`
+	ExtraData        string      `json:"extra_data"`
+	GasLimit         uint64      `json:"gas_limit"`
+	GasUsed          uint64      `json:"gas_used"`
+	BaseFeePerGas    interface{} `json:"base_fee_per_gas"`
+	Hash             string      `json:"hash"`
+	LogsBloom        string      `json:"logs_bloom"`
+	Miner            string      `json:"miner"`
+	Nonce            string      `json:"nonce"`
+	ParentHash       string      `json:"parent_hash"`
+	ReceiptRoot      string      `json:"receipt_root"`
+	Uncles           string      `json:"uncles"`
+	Size             float64     `json:"size"`
+	StateRoot        string      `json:"state_root"`
+	Timestamp        uint64      `json:"timestamp"`
+	TotalDifficulty  string      `json:"total_difficulty"`
+	TransactionsRoot string      `json:"transactions_root"`
+
+	IndexedAt string `json:"indexed_at"`
+}
+
+// Get block from database
+func (dbc *DatabaseClient) GetBlock(blockchain string, blockNumber uint64) (Block, error) {
```

Comment:
Can you create an issue which lists all functions that need to be added here? This is something that would be good for open source contributions but in general just useful to track outside of code in an issue or a spreadsheet.

Reply:
Done https://github.com/bugout-dev/moonstream/issues/568